submissions/BlueLamantine/js-dom/dom.js

```diff
@@ -0,0 +1,154 @@
+const COFFEE = [{
+	title: 'Doppio',
+	id: 'doppio',
+	classes: {
+		bgColor: 'doppio-bg',
+		textColor: 'latte-text'
+	},
+	paragraph: 'Double shot of espresso. Straight.',
+	state: true
+}, {
+	title: 'Americano',
+	id: 'americano',
+	classes: {
+		bgColor: 'americano-bg',
+		textColor: 'cappuccino-text'
+	},
+	paragraph: 'Shots of espresso diluted with water.',
+}, {
+	title: 'Macchiato',
+	id: 'macchiato',
+	classes: {
+		bgColor: 'macchiato-bg',
+		textColor: 'cappuccino-text'
+	},
+	paragraph: 'Espresso with a dash of frothy foamed milk.',
+}, {
+	title: 'Flat white',
+	id: 'flatw',
+	classes: {
+		bgColor: 'flatw-bg',
+		textColor: 'americano-text'
+	},
+	paragraph: 'Double shot of espresso with steamed milk.',
+}, {
+	title: 'Cappuccino',
+	id: 'cappuccino',
+	classes: {
+		bgColor: 'cappuccino-bg',
+		textColor: 'americano-text'
+	},
+	paragraph: 'Double shot of espresso topped with an airy thick layer of foamed milk.',
+}, {
+	title: 'Caffe latte',
+	id: 'latte',
+	classes: {
+		bgColor: 'latte-bg',
+		textColor: 'doppio-text'
+	},
+	paragraph: 'A shot of espresso in steamed milk lightly topped with foam.',
+}];
+const MENU = document.getElementById('mainMenu');
+const CUP = document.getElementById('coffeeCups');
+const ART = document.getElementById('articles');
+
+function render() {
+	document.getElementById('topHeader').innerHTML = `
+    <h1 class="mainHeader">Cat & C<span class="o"></span>ffee</h1>
+    `;
+	COFFEE.forEach(element => {
+		MENU.appendChild(listNode(element));
+		CUP.appendChild(cupNode(element));
+		ART.appendChild(contentNode(element));
+	});
+	document.getElementById('mainFooter').innerHTML = `
+    Made with &hearts; by <a href="#"> BlueLamantine </a> for <a href="#">Kottans</a>
+    `
+};
+
+function listNode(el) {
+	let listItem = document.createElement('li');
+	listItem.className = `menu-item ${el.classes.bgColor}`;
+	listItem.innerHTML = `
+    <a href="#" id="${el.id}" class="item__link ${el.classes.textColor}">${el.title}</a>
+    `;
+	if (el.state == true) {
+		listItem.classList.add('active');
+	}
+	return listItem;
+};
+
+function cupNode(el) {
+	let cup = document.createElement('div');
+	cup.id = `${el.id}-cup`;
+	cup.className = `cup ${el.id}-cup__bg `;
+	if (!el.state) {
+		cup.hidden = true;
+	}
+	cup.innerHTML = `
+        <div class = "paw-left"></div>
+        <div class = "paw-right"></div>
+    `;
+	return cup;
+}
+
+function contentNode(el) {
+	const articleContent = document.createElement('article');
+	const headerContent = document.createElement('h2');
+	const textContent = document.createElement('p');
+	articleContent.id = `${el.id}-article`;
+	articleContent.className = 'description';
+	if (!el.state) {
```

Comment:
Maybe, it's better to add some `hidden` class here and add some styles in your css for it

Reply:
There are ways to hide all unnecessary content with css, but I prefer to use HTML boolean attribute ```hidden``` in this task, with this method I hide other elements.